### DISCUSSION TITLE
How to create a baby for my breedable entity?

### DISCUSSION THREAD

**Discussion by deanwean** [FabricMC/fabric-api#2030] [Mod Dev Support] [answered]
I made my entity breedable, however when I breed them an adult entity spawns.
I want a baby entity to spawn, how do I do this? I also want the baby entity to grow up to become an adult.

This is the part of the entity code where the breeding code is.
```java
@Override
    public DotnecEntity createChild(ServerWorld world, PassiveEntity entity) {
        DotnecEntity dotentity = (DotnecEntity)BackroomsSMP.DOTNEC.create(world);
      UUID uUID = this.getOwnerUuid();
      if (uUID != null) {
        dotentity.setOwnerUuid(uUID);
        dotentity.setTamed(true);
      }

      return dotentity;
    }

    public boolean canBreedWith(AnimalEntity other) {
        if (other == this) {
           return false;
        } else if (!this.isTamed()) {
           return false;
        } else if (!(other instanceof DotnecEntity)) {
           return false;
        } else {
            DotnecEntity dotentity = (DotnecEntity)other;
           if (!dotentity.isTamed()) {
              return false;
           } else if (dotentity.isInSittingPose()) {
              return false;
           } else {
              return this.isInLove() && dotentity.isInLove();
           }
        }
     }
```

If you need me to show something more, you can ask anytime.

**Comment by warjort** [comment, marked as answer]
In general there is no one way to make a baby entity. 
Zombie, Villager and Animal babies behave/spawn in different ways.

I assume from the names of the methods you show that when you say you made your entity "breedable" you mean you implemented ````PassiveEntity````?
If you look at its subclass ````AnimalEntity```` this has a lot of the code for breeding, including a breed method that does the ````setBaby(true)```` which is the direct answer to your question.

I suspect if you try to implement breeding yourself without reference to the ````AnimalEntity```` you are going to miss lots of things like;
* spawning heart particles
* breeding cooldown
* xp for breeding
* stats for bred animals
* achievement handling
* sending entity status messages - server -> client
* etc.

To see how to use the ````AnimalEntity```` look at one of the vanilla animals.
e.g. how to add the ````AnimalMateGoal```` to your entity

**Comment by warjort** [comment]
> If you need me to show something more, you can ask anytime.

Please stop taking this attitude.

It is your responsibility to properly research a question and provide relevant information.

If somebody has to guess or play https://en.wikipedia.org/wiki/Twenty_questions
to try to get information from you, they are probably not going to bother.
Remember it is you that wants an answer to your question, you should make it easy for people to answer.

I know I feel that way with many of your questions. 
Many times (including this question) I start with something like "I will guess" when trying to answer your questions.
